packages/components/src/shared/src/useDisableInteraction.tsx

```diff
@@ -0,0 +1,23 @@
+/**
+ * useDisableInteraction provides the original event handler but disables interaction
+ * if the boolean passed is true.
+ */
+import { useCallback, MouseEventHandler } from "react";
+
+export function useDisableClick<T = Element>(disabled?: boolean, handleClick?: MouseEventHandler<T>) {
```

Comment:
Are you planning on using this hook for other components?

If not, I recommend moving the file in the `button` folder until another component needs it.